### PR TITLE
[WIP]: create resource `azuread_user_license`

### PR DESCRIPTION
**Pull request by drewmullen**
Hello! This is my first time working with the azuread provider or the go sdk yall have created to work with azure's graph api. I do have some experience writing features for the aws provider though! I'm creating this PR to show updates to this request and ask questions... Relates to issue: https://github.com/hashicorp/terraform-provider-azuread/issues/1057

Likely this will take me a while to get my arms fully around and finalize... if maintainers feel like adding commits, ill check this PR for any added commits before i make my next additions.

My plan:
 - [x] hamilton: create Create and Read functions
 - [ ] provider: create and test create/read handler 
 - [ ] hamilton: implement delete functions 
 - [ ] hamilton: add unit tests 
 - [ ] provider: create delete handler 
 - [ ] provider: create acc tests

### CODE REVIEW

File: msgraph/models.go [hunk 1]
```diff
@@ -277,6 +277,16 @@ type Application struct {
...
 }
 
+type AssignLicenseRequest struct {
+	AddLicenses *[]AddLicenses `json:"addLicenses"`
+	// Required by API but non-useful in Terraform resource construct.
+	RemoveLicenses *[]string `json:"removeLicenses"`
+}
+
+type AddLicenses struct {
+	SkuId *string `json:"skuId"`
```
Comment (manicminer) [inline msgraph/models.go:287]: Looks like this is missing the `disabledPlans` field?

**Review by manicminer — CHANGES_REQUESTED** [review]
@drewmullen Thank you for opening this PR! I've taken a look through and this is off to a good start. I've made some comments below, if you can take a look at these then I'll be happy to review again. Thanks!

**Comment by manicminer** [comment]
@drewmullen No worries, the PR will be here waiting whenever you're able to work on it :)

**Comment by manicminer** [comment]
Hi @drewmullen, just checking in to see if you might want to take another look at this? No worries if not, I'll go ahead and close it off and we can revisit again in future.

**Comment by drewmullen** [comment]
We can close this. I may come back to it someday but we found a work around

thank you for the support!

**Pull request closed without merging**